### ISSUE TITLE
[dtest] test_sequence_numbers_during_add_decommission_node crashes on `nodetool status`

### ISSUE THREAD

**Issue by fruch** [scylladb/scylladb#15182]
```
self = <alternator_stream_tests.TestAlternatorStreams object at 0x7fc215951c50>

    @pytest.mark.next_gating
    def test_sequence_numbers_during_add_decommission_node(self):
        """
        Verify shards sequence numbers on topology changes.
        1) calculate monotonic increasing sequence numbers comparing the StartingSequenceNumber of old and new shards.
        2) calculate monotonic increasing sequence numbers comparing the EndingSequenceNumber of old shards
           to EndingSequenceNumber of new shards.
        """
    
        stream_specification = enums.StreamSpecification.KEYS_ONLY.value
        self.prepare_dynamodb_cluster(num_of_nodes=4)
        node1 = self.cluster.nodelist()[0]
    
        logger.info(f'Pre setup - Creating "{TABLE_NAME}" table via "{node1.name}" node with "{stream_specification}"'
                    f' stream key')
        stream_arn = self.prefill_dynamodb_table(node=node1, stream_specification=stream_specification)[0]
        self.run_write_stress(table_name=TABLE_NAME, node=node1, num_of_item=1000, ignore_errors=True)
        decommission_thread = self.run_decommission_add_node_thread()
        streams_table = StreamsTable(stream_arn=stream_arn, dynamodb_api=self.get_dynamodb_api(node=node1))
    
        for cycle in range(3):
            logger.info(f'Starting cycle #{cycle+1}..')
            # Get original shards metadata
            original_start_sequence_numbers_set = streams_table.start_sequence_numbers_set
            original_start_sequence_numbers = streams_table.start_sequence_numbers_list
            max_original_start_sequence_number = max(
                original_start_sequence_numbers) if original_start_sequence_numbers else -1
    
            # Update table shards after topology change
>           wait_for_running_add_node(node=node1)

alternator_stream_tests.py:210: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tools/retrying.py:34: in inner
    return func(*args, **kwargs)
alternator_stream_tests.py:276: in wait_for_running_add_node
    out, err = node.nodetool('status', capture_output=True)
../scylla/.local/lib/python3.11/site-packages/ccmlib/scylla_node.py:754: in nodetool
    return super().nodetool(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ccmlib.scylla_node.ScyllaNode object at 0x7fc21420f590>, cmd = 'status'
capture_output = True, wait = True, timeout = None, verbose = True

    def nodetool(self, cmd, capture_output=True, wait=True, timeout=None, verbose=True):
        """
        Setting wait=False makes it impossible to detect errors,
        if capture_output is also False. wait=False allows us to return
        while nodetool is still running.
        When wait=True, timeout may be set to a number, in seconds,
        to limit how long the function will wait for nodetool to complete.
        """
        if capture_output and not wait:
            raise common.ArgumentError("Cannot set capture_output while wait is False.")
        env = self.get_env()
        if self.is_scylla() and not self.is_docker():
            host = self.address()
        else:
            host = 'localhost'
        nodetool = self.get_tool('nodetool')
    
        if not isinstance(nodetool, list):
            nodetool = [nodetool]
        # see https://www.oracle.com/java/technologies/javase/8u331-relnotes.html#JDK-8278972
        nodetool.extend(['-h', host, '-p', str(self.jmx_port), '-Dcom.sun.jndi.rmiURLParsing=legacy'])
        nodetool.extend(cmd.split())
        if verbose:
            self.debug(f"nodetool cmd={cmd} wait={wait} timeout={timeout}")
        if capture_output:
            p = subprocess.Popen(nodetool, universal_newlines=True, env=env, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
            stdout, stderr = p.communicate(timeout=timeout)
        else:
            p = subprocess.Popen(nodetool, env=env, universal_newlines=True)
            stdout, stderr = None, None
    
        if wait:
            exit_status = p.wait(timeout=timeout)
            if exit_status != 0:
>               raise NodetoolError(" ".join(nodetool), exit_status, stdout, stderr)
E               ccmlib.node.ToolError: Subprocess /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/share/cassandra/bin/nodetool -h 127.0.93.1 -p 7199 -Dcom.sun.jndi.rmiURLParsing=legacy status exited with non-zero status; exit status: 1; 
E               stdout: nodetool: Unable to connect to Scylla API server: java.net.ConnectException: Connection refused (Connection refused)
E               See 'nodetool help' or 'nodetool help <command>'.

../scylla/.local/lib/python3.11/site-packages/ccmlib/node.py:802: ToolError
```

crash on node-1, while node-6 is being added to the cluster:
```
ERROR 2023-08-25 15:49:50,247 [shard 0] token_metadata - get_endpoint_to_host_id_map_for_reading: endpoint 127.0.93.6 has null host_id, at: 0x5cb958e 0x5cb9b50 0x5cb9e28 0x58b9bf3 0x3d7382f 0x47f4795 0x5bd4bcc 0x5bd4b36 0x5bd4abb 0x5bd488c 0x5bdadc7 0x5bb661e 0x5bbce94 0x5bbc299 0x5bbb462 0x5bbe2b1 0x58e9eef 0x58eb127 0x58ea499 0x588da17 0x588cbcc 0x128107f 0x1282af0 0x127f60e /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x27b49 /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x27c0a 0x127d3a4
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::httpd::connection::read_one()::$_0, seastar::future<void>::then_impl_nrvo<seastar::httpd::connection::read_one()::$_0, seastar::future<void> >(seastar::httpd::connection::read_one()::$_0&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::httpd::connection::read_one()::$_0&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
   --------
   seastar::internal::do_until_state<seastar::httpd::connection::read()::$_0, seastar::httpd::connection::read()::$_1>
   --------
   seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::httpd::connection::read()::$_2, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::httpd::connection::read()::$_2>(seastar::httpd::connection::read()::$_2&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::httpd::connection::read()::$_2&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
Aborting on shard 0.
Backtrace:
  0x58d8ab8
  0x590dc22
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x3db6f
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x8e843
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x3dabd
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x2687e
  0x58b9c56
  0x3d7382f
  0x47f4795
  0x5bd4bcc
  0x5bd4b36
  0x5bd4abb
  0x5bd488c
  0x5bdadc7
  0x5bb661e
  0x5bbce94
  0x5bbc299
  0x5bbb462
  0x5bbe2b1
  0x58e9eef
  0x58eb127
  0x58ea499
  0x588da17
  0x588cbcc
  0x128107f
  0x1282af0
  0x127f60e
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x27b49
  /jenkins/workspace/scylla-master/gating-dtest-release/scylla/.ccm/scylla-repository/83ceedb18bdc33f8172ec9dc22e2eda13dae96be/libreloc/libc.so.6+0x27c0a
  0x127d3a4
```

### logs:

https://jenkins.scylladb.com/job/scylla-master/job/next/6440/artifact/dtest-raft/logs-full.release.015/1692978623941_alternator_stream_tests.py%3A%3ATestAlternatorStreams%3A%3Atest_sequence_numbers_during_add_decommission_node/

**Comment by mykaul** [comment]
@bhalevy - ping as the 1st error is around get_endpoint_to_host_id_map_for_reading()

**Comment by nyh** [comment]
@bhalevy the failure in `get_endpoint_to_host_id_map_for_reading()` :
> ERROR 2023-08-25 15:49:50,247 [shard 0] token_metadata - get_endpoint_to_host_id_map_for_reading: endpoint 127.0.93.6 has null host_id

Is a check that you added very recently in commit 44c14f3e2b7. Maybe the condition checked there is wrong? Maybe it possible that for a while, while a node is coming up, it is fine that we don't know its host_id and it looks like a null? Alternatively, maybe it's not fine - and this on_internal_error() caught a real bug?

**Comment by bhalevy** [comment]
I beat Israel to the gong :)
https://github.com/scylladb/scylladb/issues/15181

**Comment by bhalevy** [comment]
Duplicate of #15181

**Comment by bhalevy** [comment]
I came to the conclusion that the assertion was premature.
We're not there yet.
I'm testing a fix that degrades the on_internal_error to a warning as there are currently legit code paths that leave nodes in topology in normal state but with null host_id, and it is patched in only later.
Hopefully when we move gossip and token_metadata to identify nodes by host_id this will no longer be the case and we can assert the above.

**Comment by nyh** [comment]
@bhalevy can you please summarize the interesting comments on this issue (which you closed) to the issue that remained open (https://github.com/scylladb/scylladb/issues/15181)?

**Comment by bhalevy** [comment]
> @bhalevy can you please summarize the interesting comments on this issue (which you closed) to the issue that remained open (#15181)?

Sure (although it's also summarized in https://github.com/scylladb/scylladb/pull/15184)
